dmk2cw.c

```diff
@@ -908,9 +937,9 @@ main(int argc, char** argv)
 	  }
 	  if (out_fmt >= OUT_BYTES) {
 	    printf("%s%c%02x", out_fmt >= OUT_SAMPLES ? "\n" : "",
-		   "-FIAMJBX"[encoding], byte);
+		   encoding_letter[encoding], byte);
```

Comment:
Don't you have to do here `encoding_letter[encoding & ~SECTOR_END]` or am I not following something?

Reply:
Good catch. I forgot to mask off the extra flag bit in several places.
I also noticed an opportunity to clean up some code that checks if the encoding is a subtype of FM versus MFM to decide how to fill extra gap areas that aren't in the DMK file, so I changed that too. I am testing this a bit more now...

Reply:
p.s. I've done some testing of writing and reading back a couple of different kinds of disks, including the -g option and -f/-l options to force extra gap fill at the start or end of tracks. It all seems to work. Will push this commit in a second.